src/utils.rs

```diff
@@ -50,3 +58,35 @@ pub fn parse_gj_messages_response(meow: String) -> Result<Vec<GDMessage>, Generi
         })
         .collect()
 }
+
+pub fn valid_url_check(url: &str) -> bool {
+    let reg = regex::Regex::new(r#"^(http://www\.|https://www\.|http://|https://)?[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\.[a-z]{2,5}(:[0-9]{1,5})?(/.*)?$"#).unwrap();
+
+    if reg.is_match(&url.to_lowercase()) {
+        true
+    } else if reg.is_match(&url.to_lowercase()) {
+        true
+    } else {
+        println!("\x1b[0;31mInvalid URL ({})\x1b[0m", url);
```

Comment:
i'd prefer if you would use a crate like [`owo-colors`](https://crates.io/crates/owo-colors) instead of putting raw ANSI escape sequences in the string.